Fixes #480: Custom Links initialiser breaks if content_type is invalid

marcus-crane opened the pull request:
Related Issue: #480 

## New Behavior

This modification means that if a model with an invalid content type is provided, it will handle the error properly and the deployment will continue.

## Contrast to Current Behavior

The current behaviour means that if an invalid model is referenced to be associated with a custom link, then the initialization phase will error out. Upon doing so, it tries to reference the `name` field of the link object, but it assumes `link` is a Python object instead of a dict. This crashes the deployment despite this step not being required for Netbox to run.

## Discussion: Benefits and Drawbacks

The benefit of this change is that the custom links initializer shouldn't crash a deployment if the contents of the initializer yaml are invalid. As someone fumbling around with the syntax of this, it helps to know I can't break my deployment accidentally 😉 

## Changes to the Wiki

N/A

## Proposed Release Note Entry

This PR fixes a small crash that could occur with the custom links initializer if the models referenced are invalid.

## Double Check

* [x] I have read the comments and followed the PR template.
* [x] I have explained my PR according to the information in the comments.
* [x] My PR targets the `develop` branch.

marcus-crane committed:
Fix link name reference if content type is invalid

Currently, the name of a custom link is referenced
as if it is an object attribute which causes a break
due to it actually being a dictionary.

This minor change means that the startup script won't
*ahem* fail upon failing :)

Signed-Off-by: Marcus Crane <marcus.crane@daimler.com>

cimnine commented:
We don't require an issue for PRs, but we ask to fill in the PR template we provide. I believe you can copy most from the issue you've openend. Aside from that, this PR seems ready to merge. Thank you for the contribution!

marcus-crane commented:
> We don't require an issue for PRs, but we ask to fill in the PR template we provide. I believe you can copy most from the issue you've openend. Aside from that, this PR seems ready to merge. Thank you for the contribution!

Ah, apologies! I was filing the PR at night so I glossed over that part of the PR step it seems. I've updated the original PR message. I did notice that the top of the PR template mentions to use the `develop` branch while the checklist at the bottom mentions the `release` branch which is a little confusing.

Presumably that's a typo but I've changed the branch to `develop` despite `release` being the default. Let me know if you need to changed back.

marcus-crane commented:
Hmm, I also notice it seems like the builds were stalled on the day I submitted this PR so I'm not sure if they need to be restarted 🤷‍♀️
